dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -197,18 +182,26 @@ public String toJson(Contentlet con) throws DotMappingException {
 			contentletMap.put(ESMappingConstants.TITLE, con.getTitle());
 			contentletMap.put(ESMappingConstants.STRUCTURE_NAME, st.getVelocityVarName()); // marked for DEPRECATION
 			contentletMap.put(ESMappingConstants.CONTENT_TYPE, st.getVelocityVarName());
-			contentletMap.put(ESMappingConstants.STRUCTURE_TYPE, st.getStructureType() + ""); // marked for DEPRECATION
-			contentletMap.put(ESMappingConstants.BASE_TYPE, st.getStructureType() + "");
+			contentletMap.put(ESMappingConstants.STRUCTURE_TYPE, st.getStructureType()); // marked for DEPRECATION
+			contentletMap.put(ESMappingConstants.STRUCTURE_TYPE  + TEXT, st.getStructureType() + ""); // marked for DEPRECATION
```

Comment:
Let's use regular api conversion (Integer.toString) instead of these weird hacks